internal/fields/validate.go

```diff
@@ -855,6 +857,22 @@ func isArrayOfObjects(val any) bool {
 	return false
 }
 
+func isFlattenedSubfield(key string, schema []FieldDefinition) bool {
+	for strings.Contains(key, ".") {
+		i := strings.LastIndex(key, ".")
+		key = key[:i]
+		ancestor := FindElementDefinition(key, schema)
```

Comment:
Could this be replaced by findParentElementDefinition ? 

```suggestion
		ancestor := findParentElementDefinition(key, schema)
```

Reply:
mmm I guess it would be needed to somehow return also the key of the field, or being able to retrieve it from the `ancestor` struct.

Reply:
Yep, this is the problem I found, we still need to loop over the ancestors.